File: remote/src/main/java/com/kelsos/mbrc/interactors/TrackInfoInteractorImpl.java
package com.kelsos.mbrc.interactors;

import com.google.inject.Inject;
import com.kelsos.mbrc.domain.TrackInfo;
import com.kelsos.mbrc.repository.TrackRepository;
import rx.Observable;

public class TrackInfoInteractorImpl implements TrackInfoInteractor {
  @Inject private TrackRepository repository;
  @Override
  public Observable<TrackInfo> execute(boolean reload) {
    return repository.getTrackInfo(reload);
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/interactors/TrackInfoInteractorImpl.kt
package com.kelsos.mbrc.interactors

import com.kelsos.mbrc.domain.TrackInfo
import com.kelsos.mbrc.repository.TrackRepository
import rx.Observable
import javax.inject.Inject

class TrackInfoInteractorImpl
@Inject constructor(private val repository: TrackRepository) :
    TrackInfoInteractor {

  override fun load(reload: Boolean): Observable<TrackInfo> {
    return repository.getTrackInfo(reload)
  }
}
